androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java

```diff
@@ -266,255 +274,215 @@ public void test_02_create_public_link_folder_all_enabled()
 
     }
 
     /**
-     *  TEST CASE: Share public a folder and gets the link
-     *  PASSED IF: Link correctly copied in clipboard
+     *  TEST CASE: Share publicly a folder woth DownloadV/View/Upload permission
+     *  PASSED IF: Link created and visible in share view with Download/View/Upload option
+     *
      */
     @Test
-    public void test_03_get_link()
+    public void test_03_create_public_link_download_view_permission()
             throws IllegalArgumentException {
 
-        Log_OC.i(LOG_TAG, "Test Get Link Start");
-        SystemClock.sleep(WAIT_CONNECTION_MS);
+        Log_OC.i(LOG_TAG, "Test Share Public with Download/View/Upload");
+        SystemClock.sleep(WAIT_INITIAL_MS);
 
         //Select share option
-        selectShare(folder);
+        selectShare(folder2);
 
-        //Get public link
-        onView(withId(R.id.getPublicLinkButton)).perform(click());
+        //Check that no links are already created
+        onView(withId(R.id.shareNoPublicLinks)).check(matches(isDisplayed()));
+        onView(withId(R.id.addPublicLinkButton)).perform(click());
+        onView(withId(R.id.shareViaLinkNameValue)).perform(replaceText(nameShare));
+
+        //Enable the option for Download/View/Upload permissions
+        onView(withId(R.id.shareViaLinkEditPermissionReadAndWrite)).perform(click());
+
+        onView(withId(R.id.saveButton)).perform(scrollTo(),click());
         SystemClock.sleep(WAIT_CONNECTION_MS);
 
         //Check that the sharing panel is displayed
         onView(withId(R.id.parentPanel)).check(matches(isDisplayed()));
         onView(withId(R.id.alertTitle)).check(matches(isDisplayed()));
+        pressBack();
 
-        //Get text copied in the clipboard
-        onView(withText(R.string.copy_link)).perform(click());
-
-        String text = getTextFromClipboard();
-        SystemClock.sleep(WAIT_CLIPBOARD_MS);
-        //check if the copied link is correct
-        assertTrue(ERROR_MESSAGE, text.startsWith(testServerURL));
-
-        Log_OC.i(LOG_TAG, "Test Get Link Passed");
-
-    }
-
-    /**
-     *  TEST CASE: Edit the name of a public folder. Only for oC >= 10
-     *  PASSED IF: Link has the new name in share view
-     */
-    @Test
-    public void test_04_edit_name() {
 
-        Log_OC.i(LOG_TAG, "Test Edit Link Name Start");
+        //Check the name
+        onView(withText(nameShare)).check(matches(isDisplayed()));
         SystemClock.sleep(WAIT_CONNECTION_MS);
 
-        //if ownCloud >= 10, we can handle the link name. If not... skipping test.
-        if (capabilities.getVersionMayor() >= VERSION_10) {
+        //The message of "not links created yet" is gone
+        onView(withId(R.id.shareNoPublicLinks))
+                .check(matches(withEffectiveVisibility(ViewMatchers.Visibility.GONE)));
 
-            //Select share option
-            selectShare(folder);
+        onView(withId(R.id.editPublicLinkButton)).perform(click());
```

Comment:
Should we edit the share in this test or is just legacy code from when the test name was `test_04_edit_name`?